frontend/src/features/admin-form/create/builder-and-design/mutations/useDuplicateFormField.ts

```diff
@@ -60,7 +60,9 @@ export const useDuplicateFormField = () => {
         // Should not happen, should not be able to update field if there is no
         // existing data.
         if (!oldForm) throw new Error('Query should have been set')
-        oldForm.form_fields.push(newField)
+        const insertionIndex =
+          oldForm.form_fields.findIndex((o) => o._id === fieldId) + 1
```

Comment:
Oh boy, okay let me change the behaviour of this portion.

It can't exactly use `getFormFieldIndexById()` as that method is in our BE utils, and references the field as a mongoose schema. Whilst this oldForm.form_fields is from an `AdminFormDto`

This is the problem with our current implementation where both BE and FE 'assumes' the same behaviour

Reply:
Thanks for catching this Tim [abd2b64](https://github.com/opengovsg/FormSG/pull/6285/commits/abd2b649d95ace20c05f44e756172189c1881d0b)